File: Assets/ProjectGame/Lib/data/class_data_id/PlaceMap/PlaceMapRow.py
from ..BaseClassDataRow import BaseClassDataRow

class PlaceMapRow(BaseClassDataRow):
    def __init__(self):
        super().__init__()
        self.name = ""  # 名前
        self.use = False  # 使用フラグ
        self.place_map = 0  # プレスデータ（辞書）

    def read(self, reader):
        self.name = reader.read_string()
        self.use = reader.read_bool()
        self.place_map = dictionary()  # Unsupported

    @classmethod
    def from_json(cls, data: dict):
        self = cls()
        self.name = data.get('name', "")
        self.use = data.get('use', False)
        self.place_map = data.get('place_map', 0)
        return self

File: Assets/ProjectGame/Lib/data/class_data_id/PlaceMap/PlaceMapRow.js
import { BaseClassDataRow } from '../BaseClassDataRow.js';

export class PlaceMapRow extends BaseClassDataRow {
    constructor() {
        super();
        this.name = ""; // 名前
        this.use = false; // 使用フラグ
        this.place_map = 0; // プレスデータ（辞書）
    }

    read(reader) {
        this.name = reader.readString();
        this.use = reader.readBoolean();
        this.place_map = 0; // Unsupported
    }

    static fromJson(data) {
        const self = new this();
        self.name = data.name ?? "";
        self.use = data.use ?? false;
        self.place_map = data.place_map ?? 0;
        return self;
    }
}
